pkg/virt-api/webhooks/validating-webhook/admitters/pod-eviction-admitter_new_test.go

```diff
@@ -104,6 +110,149 @@ var _ = Describe("Pod eviction admitter", func() {
 			Expect(kubeClient.Fake.Actions()).To(HaveLen(1))
 		})
 	})
+
+	DescribeTable("Handle virt-launcher eviction", func(clusterWideEvictionStrategy, vmiEvictionStrategy *kvirtv1.EvictionStrategy, isVMIMigratable, expectVMIEvacuation bool) {
+		vmiOptions := []vmiOption{
+			withEvictionStrategy(vmiEvictionStrategy),
+		}
+
+		if isVMIMigratable {
+			vmiOptions = append(vmiOptions, withLiveMigratableCondition())
+		}
+
+		vmi := newVMI(testNamespace, testVMIName, testNodeName, vmiOptions...)
+
+		evictedVirtLauncherPod := newVirtLauncherPod(vmi.Namespace, vmi.Name, vmi.Status.NodeName)
+		kubeClient := fake.NewSimpleClientset(evictedVirtLauncherPod)
+
+		virtClient := kubecli.NewMockKubevirtClient(ctrl)
+		virtClient.EXPECT().CoreV1().Return(kubeClient.CoreV1()).AnyTimes()
+
+		vmiClient := kubecli.NewMockVirtualMachineInstanceInterface(ctrl)
+		virtClient.EXPECT().VirtualMachineInstance(testNamespace).Return(vmiClient).AnyTimes()
```

Comment:
Let's actually use https://github.com/kubevirt/kubevirt/pull/11495 here while we are doing such big refactor.

Reply:
This PR predates #11495 and is used in order to start the work on the descheduler integration #11532.
My plan was always to do it in a follow-up when the fake client for VMIs will be available.

Reply:
I think it is worth doing in one step otherwise it seems we do something that we will completely overhaul in next step. It should be straightforward, WDYT?

Reply:
I think we should go forward with this PR as-is in order to proceed with the descheduler integration (starting with #11532).

Replacing gomock with the vmi fake client may also require small changes to the production code which IMO deserve a dedicated PR.